src/main/java/com/exasol/errorcodecrawlermavenplugin/crawler/MitigationStepReader.java

```diff
@@ -10,22 +11,27 @@
 import spoon.reflect.code.CtInvocation;
 
 /**
- * Reader for invocations of {@link ErrorMessageBuilder#mitigation(String)}.
+ * Reader for invocations of {@link ErrorMessageBuilder#mitigation(String, Object...)}.
  */
 class MitigationStepReader implements MessageBuilderStepReader {
-    private static final String SIGNATURE = "mitigation(java.lang.String)";
+    private static final Set<String> SUPPORTED_SIGNATURE = Set.of("mitigation(java.lang.String,java.lang.Object[])",
+            "mitigation(java.lang.String)");
 
     @Override
     public void read(final CtInvocation<?> builderCall, final ErrorMessageDeclaration.Builder errorCodeBuilder,
             final Path projectDirectory) throws InvalidSyntaxException {
         final List<CtExpression<?>> arguments = builderCall.getArguments();
-        assert arguments.size() == 1;
+        assert !arguments.isEmpty();
         final CtExpression<?> messageArgument = arguments.get(0);
-        errorCodeBuilder.prependMitigation(new ArgumentReader(SIGNATURE).readStringArgumentValue(messageArgument));
+        final String mitigation = new ArgumentReader(builderCall.getExecutable().getSignature())
+                .readStringArgumentValue(messageArgument);
+        errorCodeBuilder.prependMitigation(mitigation);
+        new DirectParameterReader().readInlineParameters(arguments.size() - 1, mitigation, errorCodeBuilder);
```

Comment:
the implementation of this whole method is the very same one for `MessageStepReader`, but it calls `errorCodeBuilder.prependMitigation(mitigation);` instead of `errorCodeBuilder.message(message);`

I suggest to avoid this duplication

Reply:
I did already avoid all possible true duplication. `final CtExpression<?> messageArgument = arguments.get(0);` is not real duplication since it reads different things. (method could change but mitigation not)